source/fundamentals/crud/read-operations/retrieve.txt

```diff
@@ -180,29 +171,70 @@ to the ``FindOne()`` method, which performs the following actions:
 - Skips the first two matched documents
 
 .. io-code-block::
-   :copyable: true
+  :copyable: true
 
-   .. input::
+  .. input:: /includes/fundamentals/code-snippets/CRUD/retrieve.go
+      :start-after: begin find one docs
+      :end-before: end find one docs
       :language: go
+      :dedent:
 
-      filter := bson.D{{"date_ordered", bson.D{{"$lte", time.Date(2009, 11, 30, 0, 0, 0, 0, time.Local)}}}}
-      opts := options.FindOne().SetSkip(2)
-      
-      var result Review
-      err := coll.FindOne(context.TODO(), filter, opts).Decode(&result)
-      if err != nil {
-          panic(err)
-      }
-      
-      res, _ := json.Marshal(result)
-      fmt.Println(string(res))
-
-   .. output::
+  .. output::
       :language: none
       :visible: false
 
       {"Item":"Masala","Rating":9,"DateOrdered":"2009-11-12T05:00:00Z"}
 
+.. _golang-find-one-by-id:
+
+Find One by ObjectId Example
+````````````````````````````
+
+This example defines an ``id`` variable with a value of type ``ObjectId``
+and uses ``id`` to specify a query filter. The filter matches a document
+with an ``_id`` field value that corresponds to the ``id`` variable.
+This example queries for the following document based on its ``_id`` value:
+
+.. code-block:: json
+   :copyable: false
+   :emphasize-lines: 2
+
+    {
+        _id: ObjectId('65170b42b99efdd0b07d42de'),
+        item: "Hibiscus",
+        rating : 4,
+        date_ordered : 2009-12-18T05:00:00.000+00:00
+    }
+
+The following code passes the filter and a ``FindOneOptions`` instance
+as parameters to the ``FindOne()`` method to perform the following actions:
+
+- Match the document with the specified ``ObjectId`` value
+- Project only the ``Item`` and ``Rating`` fields of the matched document
+
+.. io-code-block::
+  :copyable: true
+
+  .. input:: /includes/fundamentals/code-snippets/CRUD/retrieve.go
+      :start-after: begin objectid
+      :end-before: end objectid
+      :language: go
+      :dedent:
+
+  .. output::
+      :language: none
+      :visible: false
+
+      {"item":"Hibiscus","rating":4}
```

Comment:
Q: this output shows the field names as lowercase, but if the code is unmarshalling the document to the `Review` type, it should look different, as it should print with the fieldnames from the struct.

Reply:
When I run the code, it outputs the lowercase field names so maybe it's not unmarshalling to a `Review`? I might be missing something in the code

Reply:
I think the field names will stay lowercase after using the MarshalExtJSON() function, unless I unmarshal again into the struct. Looks like the printed documents on[ this page](https://www.mongodb.com/docs/drivers/go/current/fundamentals/crud/read-operations/project/) also have lowercase fields.